test/integration/master/synthetic_master_test.go

```diff
@@ -224,6 +225,120 @@ func TestStatus(t *testing.T) {
 		}
 	}
 }
+func constructBody(val string, size int, field string) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(`{
+		"apiVersion": "apps/v1beta1",
+		"kind": "Deployment",
+		"metadata": {
+		   "name": "test-deployment2",
+		   "namespace": "default",
+		   `)
+
+	switch field {
+	case "labels":
+		buffer.WriteString(`"labels": {`)
+	case "annotations":
+		buffer.WriteString(`"annotations": {`)
+	case "finalizers":
+		buffer.WriteString(`"finalizers": [`)
+	}
+
+	if field == "finalizers" {
+		for i := 0; i < size; i++ {
+			buffer.WriteString("\"" + val + "\",")
+		}
+		buffer.WriteString("\"" + val + "\"]")
+	} else {
+		for i := 0; i < size; i++ {
+			buffer.WriteString("\"test" + strconv.Itoa(i) + "\" : \"" + val + "\" ,")
+		}
+		buffer.WriteString(`"test" :"a" }`)
+	}
+
+	buffer.WriteString(`
+			},
+			"spec": {
+			"replicas": 1,
+			"template": {
+			"metadata": {
+				"labels": {
+				"foo": "bar"
+				}
+			},
+			"spec": {
+				"containers": [{
+				"name": "foo",
+				"image": "foo"
+				}]
+			}
+			}
+			}
+			}`)
+
+	return buffer.String()
+}
+
+func TestObjectSizeResponses(t *testing.T) {
+	_, s, closeFn := framework.RunAMaster(nil)
+	defer closeFn()
+	transport := http.DefaultTransport
+
+	expectedMsgFor1MB := `etcdserver: request is too large`
+	expectedMsgFor2MB := `rpc error: code = ResourceExhausted desc = grpc: trying to send message larger than max`
+	expectedMsgForLargeAnnotation := `metadata.annotations: Too long: must have at most 262144 characters`
+
+	deployment1 := constructBody("a", 100000, "labels")                    // >1 MB file
```

Comment:
those numbers are hard to follow. How about calling them `deploymentMegabyteLabels`, `deploymentTwoMegabytesLabels`, ... ?